### DISCUSSION TITLE
Angular/fire(universal) automatic SSR function times out if requested by Apple devices

### DISCUSSION THREAD

**Discussion by ghost** [angular/angularfire#2714] [Q&A]
### Version info

**Angular:**
Angular version : 11.0.4

**Firebase:**
Latest
**AngularFire:**
AngularFire version 6.1.4

**Other (e.g. Ionic/Cordova, Node, browser, operating system):**
Node version: 15.4.0
### How to reproduce these conditions
Setup an Angular/fire + universal project to the latest version.
Say yes to automatic SSR function for hosting from angular/fire

**Steps to set up and reproduce**

Open website in Safari or chrome on an Apple device.
I used an iPhone 11 with iOS 14.2
And I let someone test it on macbook on the latest safari.

**Sample data and security rules**
Firebase security rules are all on Test mode and are not modified.
The default security settings apply when you setup firebase.

### Debug output

** Errors in the JavaScript console **
No JS errors in console.

** Output from `firebase.database().enableLogging(true);` **
No errors

** Screenshots **
Here is the error I get in the google cloud platform logger:
![error-1](https://user-images.githubusercontent.com/38724868/101779661-d8956500-3af5-11eb-815a-96185542f6ec.PNG)
![error-2](https://user-images.githubusercontent.com/38724868/101779681-e0eda000-3af5-11eb-8014-435db2b86eca.PNG)

### Expected behavior

The expected behavior is that the website fully renders on apple devices on first load and refresh.

### Actual behavior

The actual behavior is that sometimes the website doesn't render on first load or on refresh. However it does render and load on refresh in windows 10 and android devices.

Any help would be appreciated!

**Comment by jamesdaniels** [comment]
Since this is a very wide net, SSR & Zone.js is a deep subject with a lot of caveats, I'm going to convert to a discussion until we pin down if/where this might be an AngularFire specific bug.

**Reply by ghost** [reply]
You are right, turns out the blank page problem was due to a css issue that apple devices don't like, however the SSR timeout errors continue to happen without any clear reason as to why.
